Sources/ArcGISToolkit/Components/OfflineMapAreasView/OfflineMapAreasView.swift

```diff
@@ -0,0 +1,101 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+
+@MainActor
+public struct OfflineMapAreasView: View {
+    /// The view model for the map.
+    @State private var mapViewModel: MapViewModel
+    
+    /// The action to dismiss the view.
+    @Environment(\.dismiss) private var dismiss: DismissAction
+    
+    public init(map: Map) {
+        self.mapViewModel = MapViewModel(map: map)
+    }
```

Comment:
A few things here:

- Should mapViewModel be a state object instead of a state? Because `preplannedMapModels` is a published property that is used in the UI, I think it should be a state object. Otherwise the UI won't update.
- We don't initialize state property in initializers, as @philium said before (I cannot find the links to his explanation though) because it adversely affect the view's identity.